packages/gestalt/src/OverlayPanel/InternalOverlayPanel.js

```diff
@@ -259,23 +259,28 @@ export default function InternalOverlayPanel({
                   </Box>
                 )}
                 <ScrollBoundaryContainerProvider>
-                  <InternalScrollBoundaryContainer
-                    onScroll={updateShadows}
-                    padding={PADDING_BOINTS}
-                    ref={contentRef}
-                  >
-                    {buildDismissableSubcomponent(children)}
+                  <InternalScrollBoundaryContainer includesFooter>
+                    <ScrollBoundaryContainerProvider>
+                      <InternalScrollBoundaryContainer
+                        onScroll={updateShadows}
+                        padding={PADDING_BOINTS}
+                        ref={contentRef}
+                      >
+                        {buildDismissableSubcomponent(children)}
+                      </InternalScrollBoundaryContainer>
+                    </ScrollBoundaryContainerProvider>
+                    {Boolean(footer) && (
```

Comment:
Just wondering why the footer has to be in the scroll boundary? I think this works as expected. Does using something like 'absolute' position not work.

I noticed that the gap at the footer changed by a few pixels at the bottom. Like the old content had more spacing. Just wondering if this may cause panel stuff to cutoff?

 See the header example

After:

<img width="708" alt="image" src="https://github.com/pinterest/gestalt/assets/5509813/d7e5fa5e-a99f-4272-8b4c-4bb2ae923c13">

Before:
<img width="722" alt="image" src="https://github.com/pinterest/gestalt/assets/5509813/e9ea9061-f236-48dc-a544-06cd0964cb0f">

Reply:
I know it's just a few pixels, but just want to make sure. It may cause the UI to look not so good in places

Reply:
ScrollBoundary is going to allow POpover to just position within it, otherwise it calculates positioning based on the viewport.

Reply:
That's why without them the Popover floats when the subcontainer scrolls. We need to set the inner reference for Popover to work in that space.